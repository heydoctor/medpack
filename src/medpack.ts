import { merge } from 'lodash';
import { Configuration } from 'webpack';
import webpackMerge from 'webpack-merge';
import webpackBase from './config/webpack.base';
import webpackDev from './config/webpack.dev';
import webpackProd from './config/webpack.prod';
import defaultPaths, { IPaths } from './config/paths';
import getEnv from './config/env';

export interface IMedpackOptions {
  mode?: "development" | "production",
  publicUrl?: string,
  sourceMaps?: boolean,
  paths?: IPaths,
  env?: Object,
  webpack?: (config: Configuration) => Configuration
}

export interface IMedpack {
  config: Configuration,
  paths: IPaths,
  env: Object
}

export default ({
  mode = process.env.NODE_ENV || 'development',
  publicUrl = '',
  sourceMaps = true,
  paths: customPaths,
  env: customEnv = {},
  webpack: customWebpack = a => a,
}: IMedpackOptions = {}): IMedpack => {
  const prod = mode === 'production';
  const paths = merge(defaultPaths, customPaths);
  const env = merge(getEnv(publicUrl), customEnv)

  const options = {
    mode,
    sourceMaps,
    publicUrl,
    paths,
    env,
  };

  const baseConfig = webpackBase(options);
  const envConfig = prod ? webpackProd(options) : webpackDev(options);
  const config = customWebpack(webpackMerge(baseConfig, envConfig));

  return {
    config,
    paths,
    env,
  };
};
