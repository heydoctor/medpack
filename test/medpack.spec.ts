import medpack from '../src/medpack';

describe('medpack', () => {
  test('defaults to development', () => {
    expect(medpack()).toMatchSnapshot();
  });

  test('production', () => {
    expect(
      medpack({
        mode: 'production',
      })
    ).toMatchSnapshot();
  });
});
