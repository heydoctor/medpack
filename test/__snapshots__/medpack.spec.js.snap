// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`medpack defaults to development 1`] = `
Object {
  "config": Object {
    "devtool": "cheap-module-source-map",
    "entry": Array [
      "/Users/kyle/dev/libraries/medpack/src/config/polyfills.js",
      "/Users/kyle/dev/libraries/medpack/node_modules/react-dev-utils/webpackHotDevClient.js",
      "/Users/kyle/dev/libraries/medpack/src/index.js",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "parser": Object {
            "requireEnsure": false,
          },
        },
        Object {
          "oneOf": Array [
            Object {
              "loader": "/Users/kyle/dev/libraries/medpack/node_modules/url-loader/dist/cjs.js",
              "options": Object {
                "limit": 10000,
                "name": "static/media/[name].[hash:8].[ext]",
              },
              "test": Array [
                /\\\\\\.bmp\\$/,
                /\\\\\\.gif\\$/,
                /\\\\\\.jpe\\?g\\$/,
                /\\\\\\.png\\$/,
              ],
            },
            Object {
              "exclude": /node_modules/,
              "test": /\\(js\\|jsx\\)\\$/,
              "use": Array [
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/babel-loader/lib/index.js",
                  "options": Object {
                    "cacheDirectory": false,
                    "highlightCode": true,
                    "plugins": Array [
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-transform-destructuring/lib/index.js",
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
                        Object {
                          "loose": true,
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
                        Object {
                          "useBuiltIns": true,
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-transform-runtime/lib/index.js",
                        Object {
                          "helpers": false,
                          "polyfill": false,
                          "regenerator": true,
                        },
                      ],
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js",
                      "/Users/kyle/dev/libraries/medpack/node_modules/babel-plugin-lodash/lib/index.js",
                    ],
                    "presets": Array [
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-env/lib/index.js",
                        Object {
                          "modules": false,
                          "useBuiltIns": "entry",
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-react/lib/index.js",
                        Object {
                          "development": false,
                          "useBuiltIns": true,
                        },
                      ],
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-flow/lib/index.js",
                    ],
                  },
                },
              ],
            },
            Object {
              "exclude": /\\\\\\.module\\\\\\.css\\$/,
              "test": /\\\\\\.css\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/style-loader/index.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "importLoaders": 1,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.css\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/style-loader/index.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "camelCase": true,
                    "importLoaders": 1,
                    "localIdentName": "[name]__[local]___[hash:base64:5]",
                    "modules": true,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "exclude": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
              "test": /\\\\\\.\\(scss\\|sass\\)\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/style-loader/index.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "importLoaders": 2,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/sass-loader/lib/loader.js",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/style-loader/index.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "camelCase": true,
                    "importLoaders": 2,
                    "localIdentName": "[name]__[local]___[hash:base64:5]",
                    "modules": true,
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": false,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/sass-loader/lib/loader.js",
                  "options": Object {
                    "sourceMap": false,
                  },
                },
              ],
            },
            Object {
              "exclude": Array [
                /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
                /\\\\\\.html\\$/,
                /\\\\\\.json\\$/,
              ],
              "loader": "/Users/kyle/dev/libraries/medpack/node_modules/file-loader/dist/cjs.js",
              "options": Object {
                "name": "static/media/[name].[hash:8].[ext]",
              },
            },
          ],
        },
      ],
    },
    "optimization": Object {
      "runtimeChunk": Object {
        "name": "manifest",
      },
    },
    "output": Object {
      "chunkFilename": "static/js/[name].chunk.js",
      "devtoolModuleFilenameTemplate": [Function],
      "filename": "static/js/bundle.js",
      "pathinfo": true,
      "publicPath": "/",
    },
    "plugins": Array [
      HtmlWebpackPlugin {
        "options": Object {
          "cache": true,
          "chunks": "all",
          "chunksSortMode": "auto",
          "compile": true,
          "excludeChunks": Array [],
          "favicon": false,
          "filename": "index.html",
          "hash": false,
          "inject": true,
          "meta": Object {},
          "minify": Object {
            "collapseWhitespace": true,
            "keepClosingSlash": true,
            "minifyCSS": true,
            "minifyJS": true,
            "minifyURLs": true,
            "removeComments": true,
            "removeEmptyAttributes": true,
            "removeRedundantAttributes": true,
            "removeStyleLinkTypeAttributes": true,
            "useShortDoctype": true,
          },
          "showErrors": true,
          "template": "/Users/kyle/dev/libraries/medpack/public/index.html",
          "templateParameters": [Function],
          "title": "Webpack App",
          "xhtml": false,
        },
      },
      InterpolateHtmlPlugin {
        "replacements": Object {
          "NODE_ENV": "test",
          "PUBLIC_URL": undefined,
        },
      },
      DefinePlugin {
        "definitions": Object {
          "process.env": Object {
            "NODE_ENV": "\\"test\\"",
            "PUBLIC_URL": undefined,
          },
        },
      },
      ManifestPlugin {
        "opts": Object {
          "basePath": "",
          "fileName": "asset-manifest.json",
          "filter": null,
          "generate": null,
          "map": null,
          "publicPath": "/",
          "seed": null,
          "serialize": [Function],
          "sort": null,
          "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
          "writeToFileEmit": false,
        },
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "contextRegExp": /moment\\$/,
        "resourceRegExp": /\\^\\\\\\.\\\\/locale\\$/,
      },
      HotModuleReplacementPlugin {
        "fullBuildTimeout": 200,
        "multiStep": undefined,
        "options": Object {},
        "requestTimeout": 10000,
      },
      CaseSensitivePathsPlugin {
        "fsOperations": 0,
        "options": Object {},
        "pathCache": Object {},
        "primed": false,
      },
    ],
  },
  "env": Object {
    "NODE_ENV": "test",
    "PUBLIC_URL": undefined,
  },
  "paths": Object {
    "appBuild": "/Users/kyle/dev/libraries/medpack/build",
    "appHtml": "/Users/kyle/dev/libraries/medpack/public/index.html",
    "appIndexJs": "/Users/kyle/dev/libraries/medpack/src/index.js",
    "appNodeModules": "/Users/kyle/dev/libraries/medpack/node_modules",
    "appPackageJson": "/Users/kyle/dev/libraries/medpack/package.json",
    "appPath": "/Users/kyle/dev/libraries/medpack",
    "appPublic": "/Users/kyle/dev/libraries/medpack/public",
    "appSrc": "/Users/kyle/dev/libraries/medpack/src",
    "dotenv": "/Users/kyle/dev/libraries/medpack/.env",
    "publicUrl": undefined,
    "servedPath": "/",
    "srcPaths": Array [
      "/Users/kyle/dev/libraries/medpack/src",
    ],
    "testsSetup": "/Users/kyle/dev/libraries/medpack/src/setupTests.js",
    "useYarn": true,
  },
}
`;

exports[`medpack production 1`] = `
Object {
  "config": Object {
    "devtool": "source-map",
    "entry": Array [
      "/Users/kyle/dev/libraries/medpack/src/config/polyfills.js",
      "/Users/kyle/dev/libraries/medpack/src/index.js",
    ],
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "parser": Object {
            "requireEnsure": false,
          },
        },
        Object {
          "oneOf": Array [
            Object {
              "loader": "/Users/kyle/dev/libraries/medpack/node_modules/url-loader/dist/cjs.js",
              "options": Object {
                "limit": 10000,
                "name": "static/media/[name].[hash:8].[ext]",
              },
              "test": Array [
                /\\\\\\.bmp\\$/,
                /\\\\\\.gif\\$/,
                /\\\\\\.jpe\\?g\\$/,
                /\\\\\\.png\\$/,
              ],
            },
            Object {
              "exclude": /node_modules/,
              "test": /\\(js\\|jsx\\)\\$/,
              "use": Array [
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/babel-loader/lib/index.js",
                  "options": Object {
                    "cacheDirectory": false,
                    "highlightCode": true,
                    "plugins": Array [
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-transform-destructuring/lib/index.js",
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-class-properties/lib/index.js",
                        Object {
                          "loose": true,
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-object-rest-spread/lib/index.js",
                        Object {
                          "useBuiltIns": true,
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-transform-runtime/lib/index.js",
                        Object {
                          "helpers": false,
                          "polyfill": false,
                          "regenerator": true,
                        },
                      ],
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js",
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/plugin-proposal-export-default-from/lib/index.js",
                      "/Users/kyle/dev/libraries/medpack/node_modules/babel-plugin-lodash/lib/index.js",
                    ],
                    "presets": Array [
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-env/lib/index.js",
                        Object {
                          "modules": false,
                          "useBuiltIns": "entry",
                        },
                      ],
                      Array [
                        "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-react/lib/index.js",
                        Object {
                          "development": false,
                          "useBuiltIns": true,
                        },
                      ],
                      "/Users/kyle/dev/libraries/medpack/node_modules/@babel/preset-flow/lib/index.js",
                    ],
                  },
                },
              ],
            },
            Object {
              "exclude": /\\\\\\.module\\\\\\.css\\$/,
              "test": /\\\\\\.css\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/mini-css-extract-plugin/dist/loader.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "importLoaders": 1,
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.css\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/mini-css-extract-plugin/dist/loader.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "camelCase": true,
                    "importLoaders": 1,
                    "localIdentName": "[name]__[local]___[hash:base64:5]",
                    "modules": true,
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "exclude": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
              "test": /\\\\\\.\\(scss\\|sass\\)\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/mini-css-extract-plugin/dist/loader.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "importLoaders": 2,
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/sass-loader/lib/loader.js",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "test": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
              "use": Array [
                "/Users/kyle/dev/libraries/medpack/node_modules/mini-css-extract-plugin/dist/loader.js",
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/css-loader/index.js",
                  "options": Object {
                    "camelCase": true,
                    "importLoaders": 2,
                    "localIdentName": "[name]__[local]___[hash:base64:5]",
                    "modules": true,
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/postcss-loader/lib/index.js",
                  "options": Object {
                    "ident": "postcss",
                    "plugins": [Function],
                    "sourceMap": true,
                  },
                },
                Object {
                  "loader": "/Users/kyle/dev/libraries/medpack/node_modules/sass-loader/lib/loader.js",
                  "options": Object {
                    "sourceMap": true,
                  },
                },
              ],
            },
            Object {
              "exclude": Array [
                /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
                /\\\\\\.html\\$/,
                /\\\\\\.json\\$/,
              ],
              "loader": "/Users/kyle/dev/libraries/medpack/node_modules/file-loader/dist/cjs.js",
              "options": Object {
                "name": "static/media/[name].[hash:8].[ext]",
              },
            },
          ],
        },
      ],
    },
    "optimization": Object {
      "minimizer": Array [
        UglifyJsPlugin {
          "options": Object {
            "cache": true,
            "exclude": undefined,
            "extractComments": false,
            "include": undefined,
            "parallel": true,
            "sourceMap": true,
            "test": /\\\\\\.js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            "uglifyOptions": Object {
              "compress": Object {
                "comparisons": false,
                "ecma": 5,
                "warnings": false,
              },
              "mangle": Object {
                "safari10": true,
              },
              "output": Object {
                "ascii_only": true,
                "comments": false,
                "ecma": 5,
              },
              "parse": Object {
                "ecma": 8,
              },
            },
            "warningsFilter": [Function],
          },
        },
        OptimizeCssAssetsWebpackPlugin {
          "deleteAssetsMap": Object {},
          "options": Object {
            "assetNameRegExp": /\\\\\\.css\\$/g,
            "assetProcessors": Array [
              Object {
                "phase": "compilation.optimize-chunk-assets",
                "processor": [Function],
                "regExp": /\\\\\\.css\\$/g,
              },
            ],
            "canPrint": undefined,
            "cssProcessor": [Function],
            "cssProcessorOptions": Object {},
          },
          "phaseAssetProcessors": Object {
            "compilation.optimize-assets": Array [],
            "compilation.optimize-chunk-assets": Array [
              Object {
                "phase": "compilation.optimize-chunk-assets",
                "processor": [Function],
                "regExp": /\\\\\\.css\\$/g,
              },
            ],
            "emit": Array [],
          },
          "pluginDescriptor": Object {
            "name": "OptimizeCssAssetsWebpackPlugin",
          },
        },
      ],
      "runtimeChunk": Object {
        "name": "manifest",
      },
      "splitChunks": Object {
        "chunks": "all",
        "name": "vendors",
      },
    },
    "output": Object {
      "chunkFilename": "static/js/[name].[chunkhash:8].chunk.js",
      "devtoolModuleFilenameTemplate": [Function],
      "filename": "static/js/[name].[chunkhash:8].js",
      "path": "/Users/kyle/dev/libraries/medpack/build",
      "publicPath": "/",
    },
    "plugins": Array [
      HtmlWebpackPlugin {
        "options": Object {
          "cache": true,
          "chunks": "all",
          "chunksSortMode": "auto",
          "compile": true,
          "excludeChunks": Array [],
          "favicon": false,
          "filename": "index.html",
          "hash": false,
          "inject": true,
          "meta": Object {},
          "minify": Object {
            "collapseWhitespace": true,
            "keepClosingSlash": true,
            "minifyCSS": true,
            "minifyJS": true,
            "minifyURLs": true,
            "removeComments": true,
            "removeEmptyAttributes": true,
            "removeRedundantAttributes": true,
            "removeStyleLinkTypeAttributes": true,
            "useShortDoctype": true,
          },
          "showErrors": true,
          "template": "/Users/kyle/dev/libraries/medpack/public/index.html",
          "templateParameters": [Function],
          "title": "Webpack App",
          "xhtml": false,
        },
      },
      InterpolateHtmlPlugin {
        "replacements": Object {
          "NODE_ENV": "test",
          "PUBLIC_URL": undefined,
        },
      },
      DefinePlugin {
        "definitions": Object {
          "process.env": Object {
            "NODE_ENV": "\\"test\\"",
            "PUBLIC_URL": undefined,
          },
        },
      },
      ManifestPlugin {
        "opts": Object {
          "basePath": "",
          "fileName": "asset-manifest.json",
          "filter": null,
          "generate": null,
          "map": null,
          "publicPath": "/",
          "seed": null,
          "serialize": [Function],
          "sort": null,
          "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
          "writeToFileEmit": false,
        },
      },
      IgnorePlugin {
        "checkIgnore": [Function],
        "contextRegExp": /moment\\$/,
        "resourceRegExp": /\\^\\\\\\.\\\\/locale\\$/,
      },
      MiniCssExtractPlugin {
        "options": Object {
          "chunkFilename": "static/css/[name].[contenthash:8].chunk.css",
          "filename": "static/css/[name].[contenthash:8].css",
        },
      },
    ],
  },
  "env": Object {
    "NODE_ENV": "test",
    "PUBLIC_URL": undefined,
  },
  "paths": Object {
    "appBuild": "/Users/kyle/dev/libraries/medpack/build",
    "appHtml": "/Users/kyle/dev/libraries/medpack/public/index.html",
    "appIndexJs": "/Users/kyle/dev/libraries/medpack/src/index.js",
    "appNodeModules": "/Users/kyle/dev/libraries/medpack/node_modules",
    "appPackageJson": "/Users/kyle/dev/libraries/medpack/package.json",
    "appPath": "/Users/kyle/dev/libraries/medpack",
    "appPublic": "/Users/kyle/dev/libraries/medpack/public",
    "appSrc": "/Users/kyle/dev/libraries/medpack/src",
    "dotenv": "/Users/kyle/dev/libraries/medpack/.env",
    "publicUrl": undefined,
    "servedPath": "/",
    "srcPaths": Array [
      "/Users/kyle/dev/libraries/medpack/src",
    ],
    "testsSetup": "/Users/kyle/dev/libraries/medpack/src/setupTests.js",
    "useYarn": true,
  },
}
`;
